src/components/progress/progress.scss

```diff
@@ -0,0 +1,114 @@
+@import '../../themes/ionic.globals';
+// Progress bar
+// --------------------------------------------------
+
+/// @prop - The color of the element that is being animated
+$progress-color:rgb(16,108,200) !default;
+
+/// @prop - The background color of the progress bar
+$progress-buffer-color:  rgb(170,209,249) !default;
+
+/// @prop - The animation duration of the progress bar
+$progress-animation-duration: 2s !default;
+
+/// @prop - The speed at which the progress values transition, this is useful for determinate and buffer modes
+$progress-values-transition: 0.3s !default;
+
+/// @prop - The speed at which the buffer circles will be animating
+$progress-buffer-circle-speed: 0.3s !default;
+
+ion-progress {
+    height: 5px;
+    width: 100%;
+    position: relative;
+    overflow: hidden;
+    display: block;
+    @keyframes indeterminate {
+        0% {
+            width: 0%;
+
+            left: 0%;
+        }
+        50% {
+            left: 25%;
+
+            width: 75%;
+        }
+        75% {
+            left: 110%;
+
+            width: 0%;
+        }
+    }
+    @keyframes buffering {
+        to {
+            transform: translateX(-10px);
+        }
+    }
+
+    .animtable {
+        width: 0%;
+        height: 100%;
+        position: absolute;
+        background-color: $progress-color;
+        display: none;
+    }
+    .progress {
+        display: none;
+        height: 100%;
+        width: 0%;
+
+        background-color: $progress-color;
+        transition: width $progress-values-transition cubic-bezier(0.4,0.0,0.2,1);
+    }
+    .bufferCircles {
+        height: 4px;
+        width: 100%;
+
+        display: none;
+        mask: url("data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIj8+Cjxzdmcgd2lkdGg9IjEyIiBoZWlnaHQ9IjQiIHZpZXdQb3J0PSIwIDAgMTIgNCIgdmVyc2lvbj0iMS4xIiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciPgogIDxlbGxpcHNlIGN4PSIyIiBjeT0iMiIgcng9IjIiIHJ5PSIyIj4KICAgIDxhbmltYXRlIGF0dHJpYnV0ZU5hbWU9ImN4IiBmcm9tPSIyIiB0bz0iLTEwIiBkdXI9IjAuNnMiIHJlcGVhdENvdW50PSJpbmRlZmluaXRlIiAvPgogIDwvZWxsaXBzZT4KICA8ZWxsaXBzZSBjeD0iMTQiIGN5PSIyIiByeD0iMiIgcnk9IjIiIGNsYXNzPSJsb2FkZXIiPgogICAgPGFuaW1hdGUgYXR0cmlidXRlTmFtZT0iY3giIGZyb209IjE0IiB0bz0iMiIgZHVyPSIwLjZzIiByZXBlYXRDb3VudD0iaW5kZWZpbml0ZSIgLz4KICA8L2VsbGlwc2U+Cjwvc3ZnPgo=");
```

Comment:
If you can use `svg-background-image` instead of `mask`, please do.
If not, can you do:
`mask: url("data:image/svg+xml;charset=utf-8,#{url-encode($progress-mask-svg)}")``

Where `$progress-mask-svg` is your svg, or a scalable version of it like:
```xml
<svg xmlns='http://www.w3.org/2000/svg' viewBox="0 0 12 4">
  <ellipse cx="2" cy="2" rx="2" ry="2">
    <animate attributeName="cx" from="2" to="-10" dur="0.6s" repeatCount="indefinite" />
  </ellipse>
  <ellipse cx="14" cy="2" rx="2" ry="2" class="loader">
    <animate attributeName="cx" from="14" to="2" dur="0.6s" repeatCount="indefinite" />
  </ellipse>
</svg>
```